framework/src/problems/FEProblemBase.C

```diff
@@ -2426,12 +2453,41 @@ FEProblemBase::addMaterialHelper(std::vector<MaterialWarehouse *> warehouses,
     // If the object is boundary restricted do not create the neighbor and face objects
     if (material->boundaryRestricted())
     {
+
```

Comment:
so here I add a boundary restricted material and for interface-materials I add a special label to the warehouse, which now includes a new ocntainer